### DISCUSSION TITLE
Anpassung Modulationsgrenzen Vitocal

### DISCUSSION THREAD

**Discussion by Philip-Wiege** [open3e/open3e#112] [Q&A]
Hallo zusammen,

hat sich jemand von euch schonmal getraut, die Leistungsgrenzen für den Verdichter anzufassen?

Aktuell stört mich, dass der Verdichter im Kühlbetrieb nicht unter 27% Leistung geht. Im Heizbetrieb geht er bis 18% runter. Mich würde es nicht wundern, wenn die Werte hier jeweils 2 Byte für Min und Max-Grenze darstellen.

![image](https://github.com/open3e/open3e/assets/19406918/e74d711c-9c40-47d8-825d-ecad5f294ab9)

**Comment by philippoo66** [comment]
Moin Philip!
Sorry für die späte Reaktion! Wie du weisst bin ich ja aussen vor, aber ich meine mich zu erinnern, dass da schon mal wer was versucht hatte und nicht unbedingt weiter gekommen war (Werte wurden von der WP ignoriert). Aber vlt waren das andre dids und du hast die richtigen - probier's doch mal und berichte bitte!
Grüsse!
Phil

**Comment by Philip-Wiege** [comment]
Aktuell kann ich die Werte nicht mal auslesen wegen NRC 0x22. Entweder will die Steuerung wieder Security Access vorher oder muss in einer Art Wartungsmodus sein.

**Reply by philippoo66** [reply]
wir müssen dieses seed&key knacken! ... @abnoname  ;-)

**Comment by bu-na** [comment]
Hier mal meine decodierten DIDs (VCMU Vcal250SH SW:2404):

2630 : O3EComplexType(4, "CompressorMinMaxSpeedHeating", [O3EInt16(2, "Min", scale =10, signed = "True"), O3EInt16(2, "Max", scale =10, signed = "True")]), #250SH Unit RpS
2631 : O3EComplexType(4, "CompressorMinMaxSpeedCooling", [O3EInt16(2, "Min", scale =10, signed = "True"), O3EInt16(2, "Max", scale =10, signed = "True")]), #250SH Unit RpS
2632 : O3EComplexType(4, "CompressorMinMaxSpeedDefrost", [O3EInt16(2, "Min", scale =10, signed = "True"), O3EInt16(2, "Max")]), #250SH Unit RpS
…
admin@IoBroker02:~ $ open3e -cnfg devices.json -r 0x68c.2630,0x68c.2631,0x68c.2632 -v
0x68c 2630 CompressorMinMaxSpeedHeating {"Min": 20.0, "Max": 90.0}
0x68c 2631 CompressorMinMaxSpeedCooling {"Min": 20.0, "Max": 90.0}
0x68c 2632 CompressorMinMaxSpeedDefrost {"Min": 55.0, "Max": 100.0}
closing 0x680 - bye!
closing 0x684 - bye!
closing 0x68c - bye!
closing 0x6c3 - bye!
closing 0x6c5 - bye!
closing 0x6cf - bye!
admin@IoBroker02:~ $ 
…

Gruß Bu-Na

**Comment by Philip-Wiege** [comment]
@bu-na Interessant! Konntest du die Werte einfach so lesen ohne irgendwelche Vorbedingungen? Bleibt dein Verdichter wirklich innerhalb der Grenzen von 20 bis 90 Prozent?

**Comment by bu-na** [comment]
Die Werte sind bei mir über die ECU 0x68c ohne Vorbedingungen auslesbar und entsprechen so wie ich es beobachtet habe, dem realen Verhalten des Verdichters. Über die ECU 0x680 kommt auch bei mir 0x22 als Status. Meine VCMU hat die SW-Version 59.502.2322.20.

**Comment by Philip-Wiege** [comment]
Das nacht natürlich Sinn, dass der Parameter nur im Kältekreissteuergerät VCMU existiert! Ich habe meine mal ausgelesen. Habe noch keinen Codec eingebaut, daher nur RAW. Meiner Meinung nach kommt da bei mir aber nur Mist raus, wenn ich händisch umrechne? Wahrscheinlich mache ich hier was falsch von wegen signed / unsigned oder msb / lsb. Hier meine Daten:

<img width="731" alt="Bildschirmfoto 2024-07-18 um 15 07 04" src="https://github.com/user-attachments/assets/2dc658e7-d4c7-4e18-b784-2af14ba2c16e">

**Comment by bu-na** [comment]
2630 —> 20/110
2631 —> 30/110
2632 —> 40/80
(Bei mir Kältemittel R32, bei dir vermutlich R290; daher unterschiedliche Werte/Verdichter)

Gruß Bu-Na

**Comment by Philip-Wiege** [comment]
Das passt! Wenn ich bei der VCMU die Kompressordrehzahl abfrage, kommen da 29,9 Umdrehungen pro Sekunde raus. Da der Verdichter gerade beim Kühlen auf Min Leistung läuft, passt das zu den 30 rps laut deiner Umrechnung. Jetzt ist die große Frage, ob ich mich traue, diese Drezahl etwas abzusenken gegen Takten beim Kühlen. Im Heizbetrieb darf der Verdichter bei mir auch auf 20 rps runter. Ich sehe da eigentlich keinen Unterschied, was die Schmierung des Verdichters angeht.

**Comment by bu-na** [comment]
Die erste Frage wäre, ob der Parameter überhaupt einfach schreibbar ist. Ich könnte mir weiterhin vorstellen, dass der Kältekreisregler (bei mir ein DCI 3-4HP von Airwell) da auch ein Auge drauf hat und böse Sachen nicht zulässt. Nicht umsonst sind Kennfelder hinterlegt, die die entsprechenden Constraints definieren.
Hier mal als Beispiel ein Kennfeld eines R410-Verdichters:
![image](https://github.com/user-attachments/assets/c0ec33a5-e9b6-4861-a3c8-7632c64f9962)
Vermutlich verliert man COP…keine Ahnung….
Hier noch ein anderer Vorschlag von @abnoname 

https://community.viessmann.de/t5/Waermepumpe-Hybridsysteme/Verhindern-des-Taktens-Vorschlag-an-Viessmann/m-p/433826#M112442

Vielleicht finden wir noch die passenden Datenblätter…

**Comment by RaspiCan** [comment]
Gibt es hierzu mittlerweile neue Erkenntnisse, bei einem User wurde vom TD von Viessmann (Österreich) die Mindestleistung auf 15 rps gesetzt. Weitere User sind mir jedoch nicht bekannt.

**Reply by Philip-Wiege** [reply]
Meiner Meinung nach programmiert Viessmann die DIDs remote. Der Kältekreisregler wird per Security Access entsperrt und dann sollte der DID geschrieben werden können. Ich habe eine Möglichkeit, die passenden Keys aus den Seeds zu generieren. Ist leider nichts, was man öffentlich teilen kann. Wenn du deine Anlage modifizieren willst, kann ich unterstützen. Wir wären da aber recht weit im experimentellen Bereich unterwegs. Bei meiner Anlage will ich in Zukunft die Verdichter Minimalleistung im Kühlen absenken.

**Reply by MyHomeMyData** [reply]
@Philip-Wiege Schau bitte mal in Deinen Teams-Chat.

**Reply by propi62** [reply]
gibts neuigkeiten dazu?

**Reply by Philip-Wiege** [reply]
Noch nicht. Wir waren so verblieben, dass ich es bei meiner Anlage erstmal händisch versuche.

**Reply by robert1martin** [reply]
Ich hätte auch Interesse die Mindestleistung zu verändern... insbesondere beim kühlen die 30% Leistung bekomme ich nicht weg... Takte von unter 10min sind die Folge... so kann das nicht bleiben und kühlen ist trotz Außentemperaturen von aktuell >30°C ist wieder aus :(

@Philip-Wiege  kannst du mir hier weiter helfen? Open3E auf dem PI läuft...

**Reply by Backes89** [reply]
@robert1martin selbe Problem bei mir mit dem Kühlen. Takte sind um die 5 Minuten lang weil zu viel Leistung da ist.

**Reply by Philip-Wiege** [reply]
Um die Mindestleistung beim Kühlen abzusenken, müssten wir die minimal erlaubte Verdichterdrehzahl im Kühlbetrieb absenken. Aktuell sind 30 Hz von Viessmann einprogrammiert. Beim Heizen darf der Verdichter auf 25 Hz runter. Ich glaube laut Highly könnte man bis 10 Hz runter. Angeblich hat Viessmann bei einem User den Verdichter auf 15 Hz im Heizbetrieb runterbedatet.

Ich habe aktuell recht wenig Zeit. Bevor wir da was versuchen, müsst ihr euch im Klaren sein, dass diese Modifikation theoretisch Einfluss auf die Lebensdauer des Verdichters haben kann. Falls ihr das nicht riskieren wollt, könntet ihr auch erstmal die Unwälzpumpe auf Vollgas stellen, das hilft nochmal die Wärme besser wegzukriegen.

**Reply by robert1martin** [reply]
Ja die Risiken sind mir klar. Ich würde mich auch gern schrittweise vortasten, bis ich einen stabilen Zustand (mind. 30min Kompressor Laufzeit) erreiche. Sprich zunächst mal kühlen auch auf 25Hz und schauen wie es damit läuft, dann ggf. weiter runter. Daher wär es super zumindest mal zu wissen wie es geht, damit ich etwas experimentieren kann. Pumpen auf Volllast Probier ich sobald es wieder über 30 Grad hat. 

Würd mich freuen wenn’s die Tage mal klappt. Das Thema drängt ja nicht richte mich nach dir. Dieses WE bin ich aber auch nicht zu Hause.

**Comment by propi62** [comment]
was müsste ich bei JSON oder XML oder RAW eingeben für MIN 15 RMS?
<img width="1005" height="1102" alt="compressor" src="https://github.com/user-attachments/assets/e719d7a0-e1a4-4356-b183-bf7d9ecdd9ca" />

**Reply by MyHomeMyData** [reply]
Versuch es mal mit `{"mode":"write","data":[["2630.Min",15]],"addr":"0x68c"}`

**Reply by propi62** [reply]
zu spät. habe mail an viessmann gemacht mit der bitte den parameter umzustellen.
werde berichten wie es läuft. warte auf den nächsten Takt.

<img width="999" height="710" alt="image" src="https://github.com/user-attachments/assets/7375fa49-96d3-4376-867a-29de99326545" />

**Reply by Backes89** [reply]
Hat Viessmann dir den Wert denn schon so schnell umgestellt, dass er jetzt schon auf 15 steht oder hast du selber den Befehl abgesetzt?

**Reply by MyHomeMyData** [reply]
@propi62 Die Antwort kam eine Stunde nach der Frage und das war zu spät??? Wow.

**Reply by propi62** [reply]
hatte das mail gestern Abend 17:30 gemacht.
bis jetzt sieht es sehr gut aus.

**Comment by propi62** [comment]
Kompressor Drehzahl pendelt sich bei 15RPS ein:
<img width="1701" height="401" alt="image" src="https://github.com/user-attachments/assets/39ca3b08-4ce9-4d49-9f72-1df12f430fff" />

Werte bei 5°C Aussentemperatur  um 20:00 Uhr
<img width="632" height="70" alt="image" src="https://github.com/user-attachments/assets/a74efc55-770f-463f-9ea3-0334d886e742" />

<img width="787" height="377" alt="image" src="https://github.com/user-attachments/assets/471628a1-2f44-4b2e-af05-4000795d8789" />

Öltemperatur
<img width="1704" height="399" alt="image" src="https://github.com/user-attachments/assets/2433fc7b-f428-44a6-9bcf-750850295ace" />

Ich hatte noch nie einen so langen Takt. (7:30 h bis jetzt)
Ich hoffe nur, dass sich das nicht negativ auf die Haltbarkeit der Anlage auswirkt. (Vitocal 250-A 10kW)
Wäre toll, wenn man den Wert per open3e einstellen könnte.

**Reply by Backes89** [reply]
Wahnsinn. Wo/Wie hast du dich denn bei Viessmann dazu gemeldet das das so flott ging?

**Comment by propi62** [comment]
Wohne in der Schweiz.
Habe Mail an den Technische Support von Viessmann Schweiz gemacht:

---------------------------------------------------------------------------------------------

Guten Tag
Ich möchte Sie bitten, auf meiner Anlage den Parameter 2630 Minimaldrehzahl von 20rps auf 15rps anzupassen. (Mindestdrehzahl Kompressor)
2630:4:CompressorMinMaxSpeedHeating
 
Daten zu meiner Heizung:
Vitocal 250A
Seriennummer: #############
  
Ich würde mich sehr freuen, wenn Sie dies erledigen könnten.
Freundliche Grüsse

---------------------------------------------------------------------------------------------

Die Antwort:

---------------------------------------------------------------------------------------------

Sehr geehrter Herr ,

wir bestätigen hiermit die von Ihnen gewünschte Anpassung.

Der Parameter 2630 wurde auftragsgemäss von 20rps auf 15rps reduziert.

Freundliche Grüsse

Viessmann (Schweiz) GmbH

---------------------------------------------------------------------------------------------


Ich bin begeistert wie es seit der Umstellung läuft. Bei mir hat es wirklich viel gebracht.
PS, Das Kennfeld-Bild oben eines R410-Verdichters deckt sich genau mit dem Verhalten meiner Anlage. Sobald das Vorlaufwasser wärmer als 35 Grad = Kompressor minimum 20RPS
<img width="1692" height="395" alt="image" src="https://github.com/user-attachments/assets/f903708f-bfd9-4c76-9ae1-6e37156b60b1" />

**Reply by Philip-Wiege** [reply]
Das sieht ja wirklich gut aus! Hast du einen Wartungsvertrag mit Viessmann?

**Reply by propi62** [reply]
nein, aber es wäre schon gut, wenn man den wert auch per open3e anpassen kann. Habe gelesen, sobald die Einstellungen der Wärmepumpe zurückgesetzt würden, der Wert wieder auf 20 steht. Wie es nach einem Softwareupdate aussieht weiss ich auchnicht. Daher müsste es mal jemand ausprobieren. ;)

**Reply by propi62** [reply]
Nach dem Update auf die neuste Version bleibt der Wert 15 stehen.

**Reply by Philip-Wiege** [reply]
Danke für die Rückmeldung!

**Reply by propi62** [reply]
jetzt nach einem weiteren update auf 20.517.2629.320 wurde der Wert wieder auf 20 geändert.
Gibt es mittlerweile eine einfache Möglichkeit den Wert zu ändern. zB per SSH befehl?
sensor.2630_compressorminmaxspeedheating_min
<img width="554" height="426" alt="image" src="https://github.com/user-attachments/assets/29b1a54e-aee9-4394-aa8b-3786ef0c3a8a" />

**Comment by Philip-Wiege** [comment]
@MyHomeMyData Kann ich mit dem aktuellen "Open3E Stack" unkompliziert rohe CAN Boschaften schicken? Ich würde im ersten Schritt mal versuchen meine VCMU zu entsperren.

Ich will eigentlich noch an die Verdichter und Lüfterdrehzahlgrenzen ran...

Bin zeitlich nur nahezu voll ausgelastet, daher aktuell fast kein Input von mir.

**Reply by MyHomeMyData** [reply]
> Kann ich mit dem aktuellen "Open3E Stack" unkompliziert rohe CAN Boschaften schicken?

@Philip-Wiege Hallo Philip, verstehe die Frage nicht. Was verstehst Du unter "rohe CAN Botschaften"? Mit `cansend` kannst Du beliebige Daten an beliebige Adressen senden, das hat dann aber mit open3e nichts zu tun.

**Reply by Philip-Wiege** [reply]
Das "Problem" ist, dass ich für einen Security Access mehr als 8 Bytes versenden muss. Das macht Open3E ja über ISOTP. Ich glaube cansend kann nur ohne ISOTP bis 8 Byte senden?

**Reply by MyHomeMyData** [reply]
Ja, cansend kann nur 8 Byte bei Standard-Frames.

Hast Du schon Schreiben mit `-f77` probiert?

**Reply by Philip-Wiege** [reply]
Habe ne Möglichkeit gefunden ISOTP über die Linux Kommandozeile oder Python zu senden. Ich versuche die Tage mal, nen Seed anzufordern. Mal sehen, ob ich in ner Maximaldauer antworten muss. Wenn ja, muss ich das skriptbasiert machen.

**Comment by propi62** [comment]
Ich hab es so probiert, leider ohne Erfolg:
```
(.venv) @vitocal:~/open3e $ open3e -r 0x68C.CompressorMinMaxSpeedHeating.Min
20.0
(.venv) @vitocal:~/open3e $ open3e -w 0x68C.CompressorMinMaxSpeedHeating.Min=15.0 -f77
write f77: 1676.CompressorMinMaxSpeedHeating.Min = "15.0"
Using writeDataByIdentifier service 77. Verify the result!
[ERROR] write: Did not receive response in time. Global request timeout time has expired (timeout=3.000 sec), code: ERR/0x68e.CompressorMinMaxSpeedHeating
(.venv) @vitocal:~/open3e $ open3e -r 0x68C.CompressorMinMaxSpeedHeating.Min
20.0
```

**Reply by propi62** [reply]
niemand eine idee?

**Comment by robert1martin** [comment]
Hat mittlerweile jemand den Parameter schreiben können? 
Ich erwarte diese Woche einen Service Techniker der SW aktualisieren soll.
Den werde ich bitten mir gleich die Grenzen etwas nach unten zu korrigieren.
Aktuell sind bei mir Original
2630 20-85
2631 30-100
2632 40-80
eingestellt.

Falls also niemand weiter gekommen ist (ich nehme mal an wegen Security Access), werde ich mal einen CAN Logger an beide CANs (intern und extern) hängen und schauen was da parallel so passiert. Mit etwas Glück bekommt man ja den Key rausgerechnet.
